tests/utils/spark_operator.go

```diff
@@ -1,135 +1,158 @@
 package utils
 
 import (
+	"errors"
+	"fmt"
 	log "github.com/sirupsen/logrus"
-	coreV1 "k8s.io/api/core/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
-	"os"
-	"os/exec"
 	"strings"
+	"time"
 )
 
+const operatorDir = "../kudo-operator"
+
 type SparkOperatorInstallation struct {
-	Namespace string
-	Clients   *kubernetes.Clientset
+	Namespace    string
+	InstanceName string
+	Clients      *kubernetes.Clientset
 }
 
-func InstallSparkOperator() *SparkOperatorInstallation {
-	return InstallSparkOperatorWithNamespace(DefaultNamespace)
+func (spark *SparkOperatorInstallation) InstallSparkOperator() error {
+	return spark.InstallSparkOperatorWithParams(nil)
 }
 
-func InstallSparkOperatorWithNamespace(namespace string) *SparkOperatorInstallation {
-	clientSet, err := GetK8sClientSet()
-	if err != nil {
-		log.Fatal(err)
-		panic(err)
+func (spark *SparkOperatorInstallation) InstallSparkOperatorWithParams(params map[string]string) error {
```

Comment:
does it make sense to move `params map[string]string` into `SparkOperatorInstallation` so we have all the configuration encapsulated in one place?

Reply:
It does, thanks! I've applied your suggestion.